### PR TITLE
Implements `getTransferFee` on `IssuedCurrencyClient`

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds the method `getTransferFee` to the `IssuedCurrencyClient`, which returns the transfer fee for IOUs issued by the specified account. This is achieved by submitting an `account_info` transaction and retrieving the transfer rate from the result.

### Context of Change

Part of the IOU effort. 

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

## Before / After

`getTransferFee` exposed on `IssuedCurrencyClient`

## Test Plan

CI passes. Wrote unit tests for the function that has full codecov and passes.

### CODE REVIEW

**Comment by mvadari** [comment]
Not sure if I need more testing on this

File: src/XRP/issued-currency-client.ts [hunk 1]
```diff
@@ -348,6 +348,27 @@ export default class IssuedCurrencyClient {
...
   }
 
+  /**
+   * Set the Transfer Fees for a given issuing account.
+   * The Transfer Fee is a percentage to charge when two users transfer an issuer's IOUs on the XRPL.
+   *
+   * @see https://xrpl.org/transfer-fees.html
+   *
+   * @param address The X-address for which the transfer rate is requested.
+   * @returns A promise which resolves to a number that represents the transfer fee associated with that issuing account.
```
Comment (tedkalaw) [inline src/XRP/issued-currency-client.ts:358]: nit - `or undefined if one is not specified` or something similar

**Review by tedkalaw — APPROVED** [review]

File: test/XRP/issued-currency-client.test.ts [hunk 2]
```diff
@@ -619,6 +619,64 @@ describe('Issued Currency Client', function (): void {
     })
   })
 
+  it('getTransferFee - successful response', async function (): Promise<void> {
+    // GIVEN an IssuedCurrencyClient with mocked networking that will return a successful hash for submitTransaction
```
Comment (amiecorso) [inline test/XRP/issued-currency-client.test.ts:623]:
```suggestion
    // GIVEN an IssuedCurrencyClient with mocked networking that will successfully return information from a `getAccountInfo` call
```

File: test/XRP/issued-currency-client.test.ts [hunk 3]
```diff
@@ -619,6 +619,64 @@ describe('Issued Currency Client', function (): void {
...
+    const issuedCurrencyClient = new IssuedCurrencyClient(
+      fakeSucceedingGrpcClient,
+      fakeSucceedingWebSocketClient,
+      XrplNetwork.Test,
+    )
+
+    const expectedTransferFee = 1000000012
+
+    // WHEN setTransferFee is called
```
Comment (amiecorso) [inline test/XRP/issued-currency-client.test.ts:632]:
```suggestion
    // WHEN getTransferFee is called
```

File: test/XRP/issued-currency-client.test.ts [hunk 4]
```diff
@@ -619,6 +619,64 @@ describe('Issued Currency Client', function (): void {
...
+    const transferFee = await issuedCurrencyClient.getTransferFee(
+      this.wallet.getAddress(),
+    )
+
+    // THEN a transfer rate and is the expected value
```
Comment (amiecorso) [inline test/XRP/issued-currency-client.test.ts:637]:
```suggestion
    // THEN the expected value is returned.
```
(or something like this)

File: test/XRP/issued-currency-client.test.ts [hunk 5]
```diff
@@ -619,6 +619,64 @@ describe('Issued Currency Client', function (): void {
...
+    assert.exists(transferFee)
+    assert.equal(transferFee, expectedTransferFee)
+  })
+
+  it('getTransferFee - bad address', function (): void {
+    // GIVEN an IssuedCurrencyClient with mocked networking that will return a successful hash for submitTransaction
```
Comment (amiecorso) [inline test/XRP/issued-currency-client.test.ts:643]:
```suggestion
    // GIVEN an IssuedCurrencyClient with mocked networking that will succeed
```

File: test/XRP/issued-currency-client.test.ts [hunk 6]
```diff
@@ -619,6 +619,64 @@ describe('Issued Currency Client', function (): void {
...
+    const issuedCurrencyClient = new IssuedCurrencyClient(
+      fakeSucceedingGrpcClient,
+      fakeSucceedingWebSocketClient,
+      XrplNetwork.Test,
+    )
+
+    const badAddress = 'badAddress'
+
+    // WHEN setTransferFee is attempted THEN an error is propagated.
```
Comment (amiecorso) [inline test/XRP/issued-currency-client.test.ts:652]:
```suggestion
    // WHEN getTransferFee is attempted THEN an error is propagated.
```

File: test/XRP/issued-currency-client.test.ts [hunk 7]
```diff
@@ -619,6 +619,64 @@ describe('Issued Currency Client', function (): void {
...
+    issuedCurrencyClient.getTransferFee(badAddress).catch((error) => {
+      assert.deepEqual(error, XrpError.xAddressRequired)
+    })
+  })
+
+  it('getTransferFee - submission failure', function (): void {
+    // GIVEN an IssuedCurrencyClient which will fail to submit a transaction.
```
Comment (amiecorso) [inline test/XRP/issued-currency-client.test.ts:659]:
```suggestion
    // GIVEN an IssuedCurrencyClient with mocked networking that will fail to make a `getAccountInfo` request
```

File: test/XRP/issued-currency-client.test.ts [hunk 8]
```diff
@@ -619,6 +619,64 @@ describe('Issued Currency Client', function (): void {
...
+
+    const badAddress = 'badAddress'
+
+    // WHEN setTransferFee is attempted THEN an error is propagated.
+    issuedCurrencyClient.getTransferFee(badAddress).catch((error) => {
+      assert.deepEqual(error, XrpError.xAddressRequired)
+    })
+  })
+
+  it('getTransferFee - submission failure', function (): void {
```
Comment (amiecorso) [inline test/XRP/issued-currency-client.test.ts:658]:
```suggestion
  it('getTransferFee - request failure', function (): void {
```

File: test/XRP/issued-currency-client.test.ts [hunk 9]
```diff
@@ -619,6 +619,64 @@ describe('Issued Currency Client', function (): void {
...
+      FakeXRPNetworkClientResponses.defaultError,
+    )
+    const failingNetworkClient = new FakeXRPNetworkClient(failureResponses)
+    const issuedCurrencyClient = new IssuedCurrencyClient(
+      failingNetworkClient,
+      fakeSucceedingWebSocketClient,
+      XrplNetwork.Test,
+    )
+
+    // WHEN setTransferFee is attempted THEN an error is propagated.
```
Comment (amiecorso) [inline test/XRP/issued-currency-client.test.ts:672]:
```suggestion
    // WHEN getTransferFee is attempted THEN an error is propagated.
```

**Review by amiecorso — APPROVED** [review]
Just some test comment nits -- lmk if I'm just confused, but I think some of the verbiage could be more accurate.

**Pull request merged**